AsyncDisplayKit/ASMapNode.mm

```diff
@@ -207,7 +207,10 @@ - (void)takeSnapshot
                     [image drawAtPoint:CGPointZero];
                     
                     // Get a standard annotation view pin. Future implementations should use a custom annotation image property.
-                    MKAnnotationView *pin = [[MKPinAnnotationView alloc] initWithAnnotation:nil reuseIdentifier:@""];
+                    __block MKAnnotationView *pin;
+                    dispatch_sync(dispatch_get_main_queue(), ^{
+                      pin = [[MKPinAnnotationView alloc] initWithAnnotation:nil reuseIdentifier:@""];
+                    });
                     UIImage *pinImage = pin.image;
                     CGSize pinSize = pin.bounds.size;
```

Comment:
Technically `pin.image` and `pin.bounds` and the pin's deallocation all need to happen on the main thread also. 

Can we use `static UIImage *pinImage` and `dispatch_once` inside the main thread block to only create one pin annotation view and capture its image? We could use `pinImage.size` to have thread-safe access to the size rather than the bounds of the view.